perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_member_attribute_def_def_o365EmailAddresses_mu.java

```diff
@@ -101,13 +109,13 @@ public void checkAttributeSemantics(PerunSessionImpl sess, Member member, Attrib
 			for (Pair<Integer, Integer> memberPair : memberPairs) {
 				memberIds.add(memberPair.getLeft());
 			}
-			throw new WrongAttributeValueException(attribute, member, "some of the email addresses are already assigned to the following members: " + memberIds);
+			throw new WrongReferenceAttributeValueException(attribute, member, "some of the email addresses are already assigned to the following members: " + memberIds);
```

Comment:
If you look at this little closer, the reference attribute is the same attribute as the checked one but with different object (member). We should throw it with all information if possible.

Reply:
To do it correctly, we should probably throw an exception just for the first case of duplicity instead for all of them.

Reply:
The thrown message now contains member with whom the original member has email collision.